### PR TITLE
Add documentation on exception handling

### LINKED ISSUE #114: Documentation: retry handling
Describe retriable vs non-retriable exceptions.
Show examples of how the client application developers might want to handle retries.

### PR DESCRIPTION
**Pull request by dzou**
This adds docs about exception handling.

Fixes #114.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -226,3 +226,19 @@ The number of fragments for each row cannot be determined beforehand.
 While you can decide the number of rows you request from `SpannerResult`, the Cloud Spanner R2DBC driver will always request a fixed number of fragments from Cloud Spanner to fulfill your request and will do so repeatedly if necessary.
 
 The default number of fragments per request to Cloud Spanner is 1, but this can be configured with the `partial_result_set_fetch_size` config property for your situation.
+
+## Exception Handling
+
+The Cloud Spanner R2DBC propagates all exceptions down to the user. The exceptions thrown are split
+into two categories:
+
+- Transient: Errors caused by network problems or causes outside of the user's control.
```
Comment (meltsufin) [inline README.md]: Maybe mention the exception class here with the bullets?

File: README.md [hunk 2]
```diff
@@ -226,3 +226,19 @@ The number of fragments for each row cannot be determined beforehand.
...
+    The operations that fail due to these errors can be retried.
+    
+- Non-transient: Errors caused by invalid operations or user error. These include syntax errors,
+    invalid requests, performing invalid operations on the Spanner driver, etc. These errors
+    should not be retried.
+    
+All errors are propagated to the users as `R2dbcTransientException` and `R2dbcNonTransientException`
+respectively. The user may use reactive methods to retry operations which throw
+`R2dbcTransientException`.
```
Comment (meltsufin) [inline README.md]: Example for retrying?

File: README.md [hunk 3]
```diff
@@ -226,3 +226,19 @@ The number of fragments for each row cannot be determined beforehand.
...
+into two categories:
+
+- Transient: Errors caused by network problems or causes outside of the user's control.
+    The operations that fail due to these errors can be retried.
+    
+- Non-transient: Errors caused by invalid operations or user error. These include syntax errors,
+    invalid requests, performing invalid operations on the Spanner driver, etc. These errors
+    should not be retried.
+    
+All errors are propagated to the users as `R2dbcTransientException` and `R2dbcNonTransientException`
```
Comment (elefeint) [inline README.md]: s/as/wrapped in?

File: README.md (hunk 1 above)
Comment (dzou) [reply README.md]: Done.

File: README.md (hunk 2 above)
Comment (dzou) [reply README.md]: Done.

File: README.md (hunk 3 above)
Comment (dzou) [reply README.md]: Added sentence that exceptions are wrapped by these two classes.

**Pull request merged**